src/EventManager.php

```diff
@@ -47,24 +45,12 @@ public function dispatchEvent(string $eventName, EventArgs|null $eventArgs = nul
     /**
      * Gets the listeners of a specific event.
      *
-     * @param string|null $event The name of the event.
+     * @param string $event The name of the event.
      *
-     * @return object[]|array<string, object[]> The event listeners for the specified event, or all event listeners.
-     * @psalm-return ($event is null ? array<string, object[]> : object[])
+     * @return object[]
      */
-    public function getListeners(string|null $event = null): array
+    public function getListeners(string $event): array
     {
-        if ($event === null) {
-            Deprecation::trigger(
-                'doctrine/event-manager',
-                'https://github.com/doctrine/event-manager/pull/50',
-                'Calling %s without an event name is deprecated. Call getAllListeners() instead.',
-                __METHOD__
-            );
-
-            return $this->getAllListeners();
-        }
-
         return $this->listeners[$event];
```

Comment:
Now I'm seeing this method clearly it occurred to me :D What about an event that has no listeners? Is calling `hasListeners` the way to go? Right now this will emit PHP's warning about undefined key, which will hopefully be transformed into an exception. Maybe `?? []` should be used instead?

Reply:
Returning an empty array sounds like the reasonable thing to do.